torchx/components/component_test_base.py

```diff
@@ -15,14 +15,34 @@
 """
 
 import os
+import shutil
+import tempfile
+import time
 import unittest
 from types import ModuleType
+from typing import Any, Callable, Dict, Optional
+
+from torchx.runner import get_runner
+
+from torchx.specs import AppDef, AppStatus
 
 from torchx.specs.builders import _create_args_parser
 from torchx.specs.finder import get_component
 
 
 class ComponentTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.test_dir = tempfile.mkdtemp("torchx_component_test")
```

Comment:
I generally use TemporaryDirectory since it will clean up if the object gets destructed but in this case doesn't really matter either way